src/main/scala/models/Domain.scala

```diff
@@ -0,0 +1,30 @@
+package models
+
+import io.circe.{Codec, Decoder, Encoder}
+import io.circe.generic.semiauto.deriveCodec
+
+object Domain {
+
+  final case class CustomerId(value: String)
+
+  object CustomerId {
```

Comment:
what is the purpose of contaminating domain with library concepts like serialization?

suggestion:

don't return domain things from API, use pattern `*ApiOutput` like `CustomerApiOutput(id: String)` so you would need to have only one Codec for a concept: `Codec[CustomerApiOutput] = Codec.derive[CustomerApiOutput]`

mapping can be stupid Customer -> CustomerApiOutput that has very the same fields (it's a lot of boilerplate) but we have libraries for that: https://chimney.readthedocs.io/en/stable/supported-transformations/#into-a-case-class-or-pojo
```
case class Source(a: Int, b: Double)
case class Target(a: Int, b: Double)

Source(42, 0.07).transformInto[Target]  // == Target(42, 0.07)
Source(42, 0.07).into[Target].transform // == Target(42, 0.07)
Source(42, 0.07).transformIntoPartial[Target].asEither  // == Right(Target(42, 0.07))
Source(42, 0.07).intoPartial[Target].transform.asEither // == Right(Target(42, 0.07))
```
and to handle changes in naming:
```
(new Foo()).into[Bar].withFieldRenamed(_.getB(), _.getC).transform
```

Reply:
to do: remove that codecs completely, and do mapping in `gateway` layer for `CustomerApiOutput` that is also in `gateway` layer